cpp/src/column/column_device_view.cu

```diff
@@ -45,54 +44,75 @@ column_device_view::column_device_view( column_view source, ptrdiff_t h_ptr, ptr
                                       source.null_count(), source.offset()},
       _num_children{source.num_children()}
 {
-  if( count_descendants(source) > _num_children ) {
-    CUDF_FAIL("Columns with grand-children are not currently supported.");
-  }
-  if( _num_children > 0 )
+  size_type num_children = source.num_children();
+  if( num_children > 0 )
   {
-    column_device_view* h_column = reinterpret_cast<column_device_view*>(h_ptr);
-    column_device_view* d_column = reinterpret_cast<column_device_view*>(d_ptr);
-    d_children = d_column;
+    auto h_column = reinterpret_cast<column_device_view*>(h_ptr);
+    auto d_column = reinterpret_cast<column_device_view*>(d_ptr);
+    auto h_end = reinterpret_cast<int8_t*>(h_column + num_children);
+    auto d_end = reinterpret_cast<int8_t*>(d_column + num_children);
+    d_children = d_column; // set member ptr to device memory
     for( size_type idx=0; idx < _num_children; ++idx )
-    { // inplace-new each child
+    { // inplace-new each child into host memory
       column_view child = source.child(idx);
-      new(h_column) column_device_view(child);
-      h_column++;
+      new(h_column) column_device_view(child,reinterpret_cast<ptrdiff_t>(h_end),reinterpret_cast<ptrdiff_t>(d_end));
+      h_column++; // adv to next child
+      // update the pointers for holding this child column's child data
+      auto col_child_data_size = extent(child) - sizeof(child);
```

Comment:
If `h_end` is `column_device_view*` and `extent()` will always be a multiple of `sizeof(column_device_view)`, this could be:
```suggestion
      auto num_descendants = count_descendants(child);
      h_end += num_descendants;
      d_end += num_descendants;
```